### PR TITLE
Fix doc titles

### PR DESCRIPTION
**Pull request by markdboyd**
This is a bit of personal preference. For some reason we decided to use `hide_title: true` as a setting in the front matter (most of the time) for our docs, which means it doesn't render the `title` from the front matter on the page and we have to add our own H1 tag to the doc. Not only is there no real reason to do this, it results in extra space being rendered above the header. See the before/after pictures with the changes to use `hide_title: false` below.

But we're also not being consistent, which doesn't make the docs look great. So we should standardize one way or the other, but I see no reason to hide the title by default and then have to add one to the doc manually. There were also some inconsistencies between the front matter title in the markdown and the H1 in the markdown, which is avoided by just using the title in the front matter.

Before

docs-before

After

docs-after

### CODE REVIEW

File: docs/features/lambda_dead_letter_queue.md [hunk 1]
```diff
@@ -1,11 +1,9 @@
 ---
 id: dead_letter_queues
 title: Dead Letter Queues
-hide_title: true
+hide_title: false
 ---
 
-# Cumulus Dead Letter Queues
```
Comment (Jkovarik) [inline docs/features/lambda_dead_letter_queue.md:7]: This involves a title change.    Intentional?

**Review by Jkovarik — APPROVED** [review]
One minor comment, else this looks good, technically.      I don't really want to merge this without a quick 👍 from our PO, so @nemreid @laurenfrederick ?    Approving assuming we get a thumbs-up from them.

**Comment by nemreid** [comment]
We've talked about fixing this before and I have no issues with it as long as we maintain the titles.

File: docs/features/lambda_dead_letter_queue.md (hunk 1 above)
Comment (markdboyd) [reply docs/features/lambda_dead_letter_queue.md:7]: Yeah, intentional. The inclusion of "Cumulus" seemed superfluous and inconsistent with other docs. Let me know if that's a problem though

**Pull request merged**